core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt

```diff
@@ -300,17 +300,13 @@ class UnitMovementAlgorithms(val unit:MapUnit) {
      * DOES NOT designate whether we can reach that tile in the current turn
      */
     fun canMoveTo(tile: TileInfo): Boolean {
-        if (unit.type.isAirUnit())
-            return canAirUnitMoveTo(tile, unit)
-
-        if (!canPassThrough(tile))
-            return false
-
-        if (tile.isCityCenter() && tile.getOwner() != unit.civInfo) return false // even if they'll let us pass through, we can't enter their city
-
-        if (unit.type.isCivilian())
-            return tile.civilianUnit == null && (tile.militaryUnit == null || tile.militaryUnit!!.owner == unit.owner)
-        else return tile.militaryUnit == null && (tile.civilianUnit == null || tile.civilianUnit!!.owner == unit.owner)
+        return when {
+            unit.type.isAirUnit() -> canAirUnitMoveTo(tile, unit)
+            !canPassThrough(tile)
+                    || tile.isCityCenter() && tile.getOwner() != unit.civInfo -> false
+            unit.type.isCivilian() -> tile.civilianUnit == null && (tile.militaryUnit == null || tile.militaryUnit!!.owner == unit.owner)
+            else -> tile.militaryUnit == null && (tile.civilianUnit == null || tile.civilianUnit!!.owner == unit.owner)
+        }
```

Comment:
Please return this to its previous form, which is much easier for debugging purposes